src/test.js

```diff
@@ -36,6 +36,7 @@ export default function Test( settings ) {
 	this.module = config.currentModule;
 	this.stack = sourceFromStacktrace( 3 );
 	this.steps = [];
+	this.timeout = 0;
```

Comment:
I agree that `0` is a poor default (it should describe the lowest meaningful timeout value), but I don't mind `undefined` for "no test-specific timeout".